src/main.c

```diff
@@ -1,133 +1,105 @@
-
-  /*HD44780_Setup();
-  HD44780_PowerOn();
-  HD44780_Puts((uint8_t *)"Hello");
-  HD44780_GotoXY(2,1);
-  HD44780_Puts((uint8_t *)"World!");
-  delay(5000000);
-  HD44780_Clear();*/  
-
-
-  adc_status = adc_up();
-  if (ADC_ERROR(adc_status)) {
-    assert_failed(__FILE__, __LINE__);
-  }
-
-  __enable_irq();
-
-	while (1) {
-		GPIOB->ODR |= GPIO_MODER_MODER1_1;
-		delay(500000);
-    adc_status = adc_convert(ADC_CONVERT_PA1);
-    if (ADC_ERROR(adc_status)) {
-      assert_failed(__FILE__, __LINE__);
-    }
-    temp = adc_status.data;
-		GPIOB->ODR &= ~GPIO_MODER_MODER1_1;
-		delay(500000);
-	}
-
-  adc_status = adc_down();
-  if (ADC_ERROR(adc_status)) {
-    assert_failed(__FILE__, __LINE__);
-  }
-
-  /* TODO go to sleep
-   *
-   * Calls the ARM `WFI` instruction.
-   *
-   * WFI (Wait For Interrupt) makes the processor suspend execution (Clock is
-   * stopped) until one of the following events take place:
-   * - An IRQ interrupt
-   * - An FIQ interrupt
-   * - A Debug Entry request made to the processor.
-   */
-  // cpu_sleep();
-
-	return 0;
-}
-
-/**
-  * @brief  Reports the name of the source file and the source line number
-  *         where the assert_param error has occurred.
-  * @param  file: pointer to the source file name
-  * @param  line: assert_param error line source number
-  * @retval None
-  */
-void assert_failed(uint8_t* file, uint32_t line) {
-  /* Infinite loop */
-  while (1) {}
-}
-
-/**
-  * @}
-  */
-
-
+#include <adc.h>
+#include <gpio.h>
+#include "main.h"
+#include "config.h"
+#include "hd44780.h"
+
+struct {
+  bool paused;
+  struct {
+    uint32_t volts;
+    uint32_t millivolts;
+  } adc;
+} flextimus_prime;
+
+void assert_failed(uint8_t* file, uint32_t line) {
+  while (1) {}
+}
+
+void delay(int dly) {
+  while (dly--);
+}
+
+adc_status_t adc_convert_async_callback(adc_status_t adc_status) {
+  flextimus_prime.adc.volts = ADC_VOLTS(adc_status.data);
+  flextimus_prime.adc.millivolts = ADC_MILLIVOLTS(adc_status.data);
+  gpio_off(PAUSE_LED);
+  return ADC_OK;
+}
+
+adc_status_t adc_up_callback() {
+  adc_status_t adc_status;
+  gpio_on(PAUSE_LED);
+  adc_status = adc_convert_async(FLEX_SENSOR, adc_convert_async_callback);
+  if (ADC_ERROR(adc_status)) {
+    return adc_status;
+  }
+  return ADC_OK;
+}
+
+int main(void) {
+  adc_status_t adc_status;
+
+  gpio_up(PAUSE_LED);
+
+  adc_status = adc_up(FLEX_SENSOR, adc_up_callback);
+  if (ADC_ERROR(adc_status)) {
+    assert_failed(__FILE__, __LINE__);
+  }
+
+  __enable_irq();
+
+  //LCD Test code. NOTE: WILL HANG FOREVER IF YOU DON'T HAVE AN LCD ATTACHED
+  //TODO_MAX: Update documentation with physical set up. Replicate final schematic set up.
+
+  /*HD44780_Setup();
+  HD44780_PowerOn();
+  HD44780_Puts((uint8_t *)"Hello");
+  HD44780_GotoXY(2,1);
+  HD44780_Puts((uint8_t *)"World!");
+  delay(5000000);
+  HD44780_Clear();*/
+
+	while (1) {
+    PWR_EnterSleepMode(PWR_SLEEPEntry_WFI);
```

Comment:
This may or may not be a good idea, I really have no idea of the implications (i.e. what actually turns off and what do we have to turn back on etc, for now I think we just leave as `while (1) {}` and add this back in later